hts-precompile/IHederaTokenService.sol

```diff
@@ -390,49 +495,71 @@ interface IHederaTokenService {
     /// @param sender The sender for the transaction
     /// @param recipient The receiver of the transaction
     /// @param serialNumber The serial number of the NFT to transfer.
-    function transferNFT(address token,  address sender, address recipient, int64 serialNumber) external
-        returns (int responseCode);
-
-    /// Allows spender to withdraw from your account multiple times, up to the value amount. If this function is called 
+    function transferNFT(
+        address token,
+        address sender,
+        address recipient,
+        int64 serialNumber
+    ) external returns (int responseCode);
+
+    /// Allows spender to withdraw from your account multiple times, up to the value amount. If this function is called
     /// again it overwrites the current allowance with value.
     /// Only Applicable to Fungible Tokens
     /// @param token The hedera token address to approve
     /// @param spender the account address authorized to spend
     /// @param amount the amount of tokens authorized to spend.
     /// @return responseCode The response code for the status of the request. SUCCESS is 22.
-    function approve(address token, address spender, uint256 amount) external returns (int responseCode);
+    function approve(
+        address token,
+        address spender,
+        uint256 amount
+    ) external returns (int responseCode);
```

Comment:
The [IERC20](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/8b778fa20d6d76340c5fac1ed66c80273f05b95a/contracts/token/ERC20/IERC20.sol#L66) approve function also returns a boolean. Also what meaning would the response codes have in the context of ERC functions called on HTS tokens? So far we were strictly following the IERC20 and IERC721 interface definitions for function return types.

Reply:
In the context of ERC there wouldn't be an equivalent but on the Hedera debugging side we want to provide further context in the case of operation failure.
For strict IERC20/IERC721 function would it break to add the response code as an additional return?

Reply:
It won't be possible to add a response code as a return type when using the IERC20/721 interfaces to make the call like this `IERC721(token).approve(to, tokenId)` because it would break the function selector.
But for the newly added functions e.g. `approve(address token, address to, uint256 tokenId)` I think it would be fine to add a response code. Probably we should add the boolean response as well though, since people familiar with ERC20 may expect it.

Reply:
would anything other than a `ResponseEnum.SUCCESS` be a `false`?  Can it be a success response code and still fail?

Reply:
I don't think so since the nuance of other response code highlight where the transaction was partially handled but an undesired side effect or context was encountered.
@stoqnkpL please confirm

Reply:
Yes, it would either be successful or return a different response code than `SUCCESS`.